### PR TITLE
feat: added log file streaming

### PR DESCRIPTION
**Pull request by aarontravass**
As stated in #2 , this configuration will output logs to a file.

### CODE REVIEW

File: src/filelogger.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  writeFileSync,
+  createWriteStream,
+  WriteStream,
+  mkdirSync
+} from 'fs'
+import { dirname as directoryname } from 'path'
+
+export class FileLogger {
+  private readonly filename: string
+  private readonly dirname: string
```
Comment (v1rtl) [inline src/filelogger.ts]: use private property methods instead, aka `#dirname`

File: src/filelogger.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  constructor(filename: string) {
+    this.dirname = directoryname(filename)
+    this.filename = filename
+    this._stat()
+    this._createWritableStream()
+  }
+
+  private _stat() {
+    //check if file exists
+    if (existsSync(this.filename)) {
```
Comment (v1rtl) [inline src/filelogger.ts]: `existsSync` is deprecated, you should use `fs.access` API with a special code, I think it's `F_OK`

**Review by v1rtl — CHANGES_REQUESTED** [review]

**Comment by aarontravass** [comment]
I was getting this error during build
```typescript
> @tinyhttp/logger@1.3.4 build C:\Users\aaron\Documents\projects\logger
> rollup -c

(node:33272) ExperimentalWarning: Importing JSON modules is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

src/index.ts → dist...
(!) Plugin typescript: @rollup/plugin-typescript TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.
src/filelogger.ts: (9:5)

9     this.#dirname = directoryname(filename)
      ~~~~~~~~~~~~~

(!) Plugin typescript: @rollup/plugin-typescript: Rollup 'sourcemap' option must be set to generate source maps.
(!) Plugin typescript: @rollup/plugin-typescript: outputToFilesystem option is defaulting to true.
(!) Unresolved dependencies
https://rollupjs.org/troubleshooting/#warning-treating-module-as-external-dependency
fs (imported by "src/filelogger.ts")
path (imported by "src/filelogger.ts")
created dist in 2.3s
```
hence I added tslib

File: package.json [hunk 3]
```diff
@@ -54,9 +54,11 @@
     "eslint-plugin-prettier": "^4.2.1",
     "expect": "^29.5.0",
     "husky": "^8.0.3",
+    "install": "^0.13.0",
```
Comment (v1rtl) [inline package.json]: what's this dep?

**Review by v1rtl — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (aarontravass) [reply package.json]: got installed accidentally. weird.

**Comment by aarontravass** [comment]
coverage is not stopping. hmm

**Comment by v1rtl** [comment]
@aarontravass there's a permission error for reading a file, please check your tests and make sure they don't reference any system files

**Comment by aarontravass** [comment]
> @aarontravass there's a permission error for reading a file, please check your tests and make sure they don't reference any system files

it's a problem even on master https://github.com/tinyhttp/logger/actions/runs/5100346720/jobs/9168579370

**Comment by aarontravass** [comment]
Let me take a look once again.

**Comment by aarontravass** [comment]
ci passed :)

**Pull request merged**